include/circt/Dialect/LLHD/IR/SignalOps.td

```diff
@@ -0,0 +1,248 @@
+  let description = [{
+    This instruction represents a storage element. It drives its output onto
+    the 'signal' value. An arbitrary amount of triggers can be added to the
+    storage element. However, at least one is required. They are quadruples
+    consisting of the new value to be stored if the trigger applies, the
+    mode and trigger value which specify when this trigger has to be applied
+    as well as a delay. Optionally, each triple may also have a gate
+    condition, in this case the trigger only applies if the gate is one. If
+    multiple triggers apply the left-most in the list takes precedence.
+
+    There are five modes available:
+
+    | Mode   | Meaning                                                         |
+    |--------|-----------------------------------------------------------------|
+    | "low"  | Storage element stores `value` while the `trigger` is low. Models active-low resets and low-transparent latches.
+    | "high" | Storage element stores `value` while the `trigger` is high. Models active-high resets and high-transparent latches.
+    | "rise" | Storage element stores `value` upon the rising edge of the `trigger`. Models rising-edge flip-flops.
+    | "fall" | Storage element stores `value` upon the falling edge of the `trigger`. Models falling-edge flip-flops.
+    | "both" | Storage element stores `value` upon the a rising or a falling edge of the `trigger`. Models dual-edge flip-flops.
+
+    This instruction may only be used in an LLHD entity.
+
+    Syntax:
+
+    ```
+    reg-op ::= `llhd.reg` signal-ssa-value ( `,` `(` value-ssa-value `,` mode-string trigger-ssa-value `after` delay-ssa-value ( `if` gate-ssa-value )? `:` value-type )+ attr-dict `:` signal-type
+    ```
+
+    Examples:
+
+    A rising, falling, and dual-edge triggered flip-flop:
+
+    ```mlir
+    llhd.reg %Q, (%D, "rise" %CLK after %T : !llhd.sig<i8>) : !llhd.sig<i8>
+    llhd.reg %Q, (%D, "fall" %CLK after %T : !llhd.sig<i8>) : !llhd.sig<i8>
+    llhd.reg %Q, (%D, "both" %CLK after %T : !llhd.sig<i8>) : !llhd.sig<i8>
+    ```
+
+    A rising-edge triggered flip-flop with active-low reset:
+
+    ```mlir
+    llhd.reg %Q, (%init, "low" %RSTB after %T : !llhd.sig<i8>), (%D, "rise" %CLK after %T : !llhd.sig<i8>) : !llhd.sig<i8>
+    ```
+
+    A rising-edge triggered enable flip-flop with active-low reset:
+
+    ```mlir
+    llhd.reg %Q, (%init, "low" %RSTB after %T : !llhd.sig<i8>), (%D, "rise" %CLK after %T if %EN : !llhd.sig<i8>) : !llhd.sig<i8>
+    ```
+
+    A transparent-low and transparent-high latch:
+
+    ```mlir
+    llhd.reg %Q, (%D, "low" %CLK after %T : !llhd.sig<i8>) : !llhd.sig<i8>
+    llhd.reg %Q, (%D, "high" %CLK after %T : !llhd.sig<i8>) : !llhd.sig<i8>
+    ```
+
+    An SR latch:
+
+    ```mlir
+    %0 = llhd.const 0 : i1
+    %1 = llhd.const 1 : i1
+    llhd.reg %Q, (%0, "high" %R after %T : !llhd.sig<i1>), (%1, "high" %S after %T : !llhd.sig<i1>) : !llhd.sig<i1>
```

Comment:
plz wrap to 80 columns

Reply:
Sorry, assumed code blocks shouldn't be wrapped